components/decimal/src/format.rs

```diff
@@ -23,43 +28,74 @@ pub struct FormattedFixedDecimal<'l> {
 
 impl FormattedFixedDecimal<'_> {
     /// Returns the affixes needed for the current sign, as (prefix, suffix)
-    fn get_affixes(&self) -> Option<(&str, &str)> {
+    fn get_affixes(&self) -> Option<(Part, (&str, &str))> {
         match self.value.sign() {
             Sign::None => None,
-            Sign::Negative => Some(self.symbols.minus_sign_affixes()),
-            Sign::Positive => Some(self.symbols.plus_sign_affixes()),
+            Sign::Negative => Some((parts::MINUS_SIGN, self.symbols.minus_sign_affixes())),
+            Sign::Positive => Some((parts::PLUS_SIGN, self.symbols.plus_sign_affixes())),
         }
     }
 }
 
 impl Writeable for FormattedFixedDecimal<'_> {
```

Comment:
Note: Obviously this is a hot path that we should try to keep optimized. If you see a better way for me to write this logic, please suggest it.